src/solver/highlevel.jl

```diff
@@ -64,11 +64,11 @@ end
 
 # Singular Value Decomposition
 
-struct CuSVD{T,Tr,M<:AbstractArray{T}} <: LinearAlgebra.Factorization{T}
+struct CuSVD{T,Tr,M<:CuMatrix{T}} <: LinearAlgebra.Factorization{T}
```

Comment:
What I had in mind was
```julia
struct CuSVD{T,Tr} <: LinearAlgebra.Factorization{T}
    U::CuMatrix{T}
    S::CuVector{Tr}
    Vt::CuMatrix{T}
end
```
since then there would be only two type parameters.